Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4131,3 +4131,127 @@ WebKitFeatureList* webkit_settings_get_development_features(void)
+        }
+
+        GValue value = G_VALUE_INIT;
+        bool isValueSet = false;
+        switch (G_PARAM_SPEC_VALUE_TYPE(properties.get()[i])) {
+        case G_TYPE_BOOLEAN: {
+            bool boolValue = g_key_file_get_boolean(keyFile, groupName, name, &lookupError.outPtr());
+            if (!boolValue && lookupError) {
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            g_value_init(&value, G_TYPE_BOOLEAN);
+            g_value_set_boolean(&value, boolValue);
+            isValueSet = true;
+            break;
+        }
+        case G_TYPE_UINT: {
+            guint64 uintValue = g_key_file_get_uint64(keyFile, groupName, name, &lookupError.outPtr());
+            if (!uintValue && lookupError) {
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            if (uintValue > G_MAXUINT) {
+                g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE, "Value for '%s' exceeds maximum integer size", name);
+                return FALSE;
+            }
+            g_value_init(&value, G_TYPE_UINT);
+            g_value_set_uint(&value, static_cast<guint>(uintValue));
+            isValueSet = true;
+            break;
+        }
+        case G_TYPE_STRING: {
+            GUniquePtr<char> stringValue(g_key_file_get_string(keyFile, groupName, name, &lookupError.outPtr()));
+            if (!stringValue) {
+                g_assert(lookupError);
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            g_value_init(&value, G_TYPE_STRING);
+            g_value_take_string(&value, stringValue.release());
+            isValueSet = true;
+            break;
+        }
+        default:
+            break;
+        }
+        if (isValueSet) {
+            g_ptr_array_add(propertyNames.get(), const_cast<char*>(name));
+            g_array_append_val(values.get(), value);
+        }
+    }
+
+    size_t length;
+    GUniqueOutPtr<GError> getKeysError;
+    GUniquePtr<char*> allKeys(g_key_file_get_keys(keyFile, groupName, &length, &getKeysError.outPtr()));
+    if (UNLIKELY(getKeysError)) {
+        g_propagate_error(error, getKeysError.release());
+        return FALSE;
+    }
+
+    for (unsigned i = 0; i < length; i++) {
+        auto key = allKeys.get()[i];
+        if (!g_ptr_array_find_with_equal_func(propertyNames.get(), static_cast<gconstpointer>(key), g_str_equal, nullptr)) {
+            g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE, "The %s group contains an invalid setting: %s", groupName, key);
+            return FALSE;
+        }
+    }
+
+    g_object_freeze_notify(G_OBJECT(settings));
+    g_object_setv(G_OBJECT(settings), propertyNames->len, (const char**)propertyNames->pdata, reinterpret_cast<GValue*>(values->data));
+    g_object_thaw_notify(G_OBJECT(settings));
```

Comment:
`g_object_setv` already does the freeze/thaw, so we don't need to do it explicitly.

Reply:
Use C++ style cast too.

Reply:
Couldn't make C++ casts work :(